libs/libcommon/src/libcommon/viewer_utils/asset.py

```diff
@@ -37,6 +34,13 @@ class AudioSource(TypedDict):
     type: str
 
 
+def generate_asset_src(
+    base_url: str, dataset: str, revision: str, config: str, split: str, row_idx: int, column: str, filename: str
+) -> tuple[str, str]:
+    dir_path = f"{parse.quote(dataset)}/{DATASET_SEPARATOR}/{revision}/{DATASET_SEPARATOR}/{parse.quote(config)}/{parse.quote(split)}/{str(row_idx)}/{parse.quote(column)}"
```

Comment:
what does it mean for dataset="asoria/mydataset" for example? I imagine that the slash is quoted. Is that what we want?

Reply:
I think by default, "/" won't be quoted https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote but I can set a specific `safe` parameters to prevent.

Reply:
currently we keep it, and it creates a subdirectory. Do we want to stick with this?

Reply:
I think it is not a big deal, only existing cached assets would be affected (Because /rows, /filter, and /search will upload image/audio with the new path).
 Do you find any inconvenience with the change @lhoestq?